File: LeetCode/0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findMedianSortedArrays = function(nums1, nums2) {
    var m = nums1.length, n = nums2.length;
    var p1 = 0, p2 = 0;
    if (m === 0) {
        return n % 2 === 0 ? (nums2[Math.floor(n/2)-1]+nums2[Math.floor(n/2)])/2 : nums2[Math.floor(n/2)];
    } else if (n === 0) {
        return m % 2 === 0 ? (nums1[Math.floor(m/2)-1]+nums1[Math.floor(m/2)])/2 : nums1[Math.floor(m/2)];
    }

    function find_min(a, b) {
        if (p1 < nums1.length && p2 < nums2.length) {
            return nums1[p1] < nums2[p2] ? nums1[p1++] : nums2[p2++];
        } else if (p1 < nums1.length) {
            return nums1[p1++];
        } else if (p2 < nums2.length) {
            return nums2[p2++];
        }
        return -1;
    }

    for (let i = 0; i < (m+n)/2-1; i++) {
        find_min(nums1, nums2);
    }
    if ((m+n)%2 === 0) {
        return (find_min(nums1, nums2)+find_min(nums1, nums2))/2
    } else {
        return find_min(nums1, nums2);
    }
};

File: LeetCode/0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        ptr1, ptr2 = 0, 0
        mergedArray = []
        if len(nums1) == 0 or len(nums2) == 0:
            mergedArray = nums1 + nums2
        else:
            while True:
                if nums1[ptr1] > nums2[ptr2]:
                    mergedArray.append(nums2[ptr2])
                    ptr2 += 1
                    if ptr2 == len(nums2):
                        mergedArray += nums1[ptr1:]
                        break
                else:
                    mergedArray.append(nums1[ptr1])
                    ptr1 += 1
                    if ptr1 == len(nums1):
                        mergedArray += nums2[ptr2:]
                        break
        if len(mergedArray) % 2 == 0:
            return (mergedArray[len(mergedArray)//2-1]+mergedArray[len(mergedArray)//2])/2
        else:
            return mergedArray[len(mergedArray)//2]